### ISSUE TITLE
Support for Entity Framework 6 (.net framework)

### ISSUE THREAD

**Issue by arturtadel** [aws/aws-xray-sdk-dotnet#160]
Hi!

is there is any possibility to integrate X-Ray with Entity Framework 6 (not .net Core)?
do you have any solution "how to" do that or you have any plans to prepare similar integration as for .net Core?

**Comment by lupengamzn** [comment]
Hey @arturtadel ,

So far we haven't have a plan to support that, but I can definitely add this to our backlog. The solution to integrate EF 6 is similar to EF Core, which only needs to implement [IDbCommandInterceptor](https://docs.microsoft.com/en-us/dotnet/api/system.data.entity.infrastructure.interception.idbcommandinterceptor?view=entity-framework-6.2.0) and initialize it in your application. Currently the [.NET agent](https://github.com/aws/aws-xray-dotnet-agent/blob/main/src/sdk/EntityFramework/EntityFrameworkHandler.cs) supports integration with EF 6 (.NET framework).

**Comment by arturtadel** [comment]
I've added a default interceptor similar to in .NET agent. but now when `SqlRequestUtil.BeginSubsegment(command);` is called I can see an error `Entity doesn't exists in HTTPContext`

![image](https://user-images.githubusercontent.com/24475528/98561726-85878280-22a9-11eb-87bd-3822e4493212.png)

**Comment by lupengamzn** [comment]
Hey @arturtadel ,

The reason why you received this error is because the subsegment can't find its parent segment. Have you instrumented your application with [X-Ray](https://github.com/aws/aws-xray-sdk-dotnet/tree/master#aspnet-framework-net--nuget) handler to generate segment before any subsegment is created?

**Comment by arturtadel** [comment]
Hi @lupengamzn,
Yes, I've overridden the Init() function in my Global.asax.cs. and before I'm added the interceptor, I was able to trace all requests through X-Ray.

**Comment by lupengamzn** [comment]
May I know where did you bind your EF 6 interceptor, which may be bonded multiple times and causing this error. Also, a repro will be very helpful if you can provide one.

**Comment by arturtadel** [comment]
unfortunately, I can't provide you with the repository.

I will paste some parts of the code here
`Init` function from `Global.asax.cs`
`        public override void Init()
        {
            base.Init();
            AWSXRayASPNET.RegisterXRay(this, "my application");
        }`
for binding Interceptor I've tried few options:
1. in `web.config` (without changes in my `EFConfiguration.cs`)
`<entityFramework codeConfigurationType="MyProject.EFConfiguration, MyProject">
    <interceptors>
        <interceptor type="MyProject.MyEFInterceptor, MyProject">
        </interceptor>
    </interceptors>
</entityFramework>`

2. from code. in my `EFConfiguration` I've added `this.AddInterceptor(new MyEFInterceptor())` and in my `web.config` I've added only
`<entityFramework codeConfigurationType="MyProject.EFConfiguration, MyProject"></entityFramework>`

3. I've removed `codeConfigurationType="MyProject.EFConfiguration, MyProject"` from my `web.config` and for each dbContext I've added attribute like `[DBConfigurationType(typeof(EFConfiguration))]` but this causes other errors.

4. I've tried to concatenate both 1 and 2 with the same error.
